munit/shared/src/main/scala/hedgehog/munit/HedgehogAssertions.scala

```diff
@@ -0,0 +1,398 @@
+  * (which should be rare in user code), this should result in fairly easy
+  * adoption by munit and munit-scalacheck users.
+  */
+trait HedgehogAssertions { self: Assertions =>
+
+  /** Turns off hedgehog munit-like assertions, so users can use both property-
+    * and non-property- based-tests in their test suites. By using the passed
+    * assertions parameter, all the standard munit assertions that do not return
+    * unit are available, despite name ambiguities.
+    *
+    * ==Usage==
+    * {{{
+    *   test("1 + 1 is 2"){
+    *     withMunitAssertions{ assertions =>
+    *       assertEquals(1 + 1, 2)
+    *     }
+    *   }
+    * }}}
+    *
+    * @param body
+    *   a test body, taking an assertions parameter
+    * @return
+    */
+  def withMunitAssertions(body: => Assertions => Any): Any = body(
+    this.asInstanceOf[Assertions]
+  )
+
+  /** @see
+    *   [[hedgehog.core.Result.Failure]]
+    */
+  type Failure = Result.Failure
+
+  /** @see
+    *   [[hedgehog.core.Result.Success]]
+    */
+  type Success = Result.Success.type
+
+  /** Alias for Result.Success
+    *
+    * @see
+    *   [[hedgehog.core.Result.Success]]
+    */
+  lazy val Success: Success = Result.Success
+
+  /** Alias for Result.Failure
+    *
+    * @see
+    *   See [[hedgehog.core.Result.Failure]]
+    */
+  def Failure(log: List[hc.Log]) = Result.Failure(log)
+
+  /** Alias for Result.success
+    *
+    * @see
+    *   [[hedgehog.core.Result.success]]
+    */
+  def success = Result.success
+
+  /** Alias for Result.failure
+    *
+    * @see
+    *   [[hedgehog.core.Result.failure]]
+    */
+  def failure = Result.failure
+
+  /** Alias for Result.error
+    *
+    * @see
+    *   [[hedgehog.core.Result.error]]
+    */
+  def error(e: Exception) = Result.error(e)
+
+  /** Alias for Result.all
+    *
+    * @see
+    *   [[hedgehog.core.Result.all]]
+    */
+  def all(l: List[Result]) = Result.all(l)
+
+  /** Alias for Result.any
+    *
+    * @see
+    *   [[hedgehog.core.Result.any]]
+    */
+  def any(l: List[Result]) = Result.any(l)
+
+  /** Alias for Result.diff
+    *
+    * @see
+    *   [[hedgehog.core.Result.diff]]
+    */
+  def diff[A, B](a: A, b: B)(f: (A, B) => Boolean) = Result.diff(a, b)(f)
+
+  /** Alias for Result.diffNamed
+    *
+    * @see
+    *   [[hedgehog.core.Result.diffNamed]]
+    */
+  def diffNamed[A, B](logName: String, a: A, b: B)(f: (A, B) => Boolean) =
+    Result.diffNamed(logName, a, b)(f)
+
+  /** Fails the test with a failure Result when `cond` is `false`.
+    *
+    * Analagous to munit.Assertions.assert.
+    *
+    * Only the condition is used. Clues are ignored.
+    *
+    * @param cond
+    * @param clue
+    *   ignored -- usage triggers the deprecation warning
+    * @return
+    *   Success iff cond is true. Failure otherwise.
+    */
+  @deprecated(
+    "Clues are unnecessary with hedgehog. Use HedgehogAssertions.diff, which will automatically output clues",
+    ""
+  )
```

Comment:
I'm just wondering why it has these `@deprecated` ones. Is it to have the same methods as munit's?

Reply:
Yes, it's for munit assertion compatibility. I also wanted some way to encourage users to use the diff assertions instead.

I don't think it's a highly-used feature,  but one of my goals for creating this was to also create a scalafix to do automatic rewriting for my private projects that are on munit and scalacheck. 

This is slightly easier to do if the interfaces taking clues exist; but I could just as well implement that in the migration I'm writing, so if we want them removed from the pr, it's fine with me.

Reply:
Alright, thanks for the explanation. All good. I was just curious. You know well about munit so I think I should let you decide.

Reply:
Yeah, I think I'll leave them then.